test/wallet/index.ts

```diff
@@ -202,6 +214,252 @@ describe('Wallet', function () {
       // TODO: Check the output of the signature against a known result
       assert.isString(signedTx)
     })
+
+    it('sign', async function () {
+      const result = wallet2.signTransaction(
+        JSON.parse(REQUEST_FIXTURES.normal.txJSON) as unknown as Transaction,
+      )
+      assert.deepEqual(result, RESPONSE_FIXTURES.normal.signedTransaction)
+    })
+
+    it('sign with lowercase hex data in memo (hex should be case insensitive)', async function () {
+      const secret = 'shd2nxpFD6iBRKWsRss2P4tKMWyy9'
+      const lowercaseMemoTx: Transaction = {
+        TransactionType: 'Payment',
+        Flags: 2147483648,
+        Account: 'rwiZ3q3D3QuG4Ga2HyGdq3kPKJRGctVG8a',
+        Amount: '10000000',
+        LastLedgerSequence: 14000999,
+        Destination: 'rUeEBYXHo8vF86Rqir3zWGRQ84W9efdAQd',
+        Fee: '12',
+        Sequence: 12,
+        SourceTag: 8888,
+        DestinationTag: 9999,
+        Memos: [
+          {
+            Memo: {
+              MemoType:
+                '687474703a2f2f6578616d706c652e636f6d2f6d656d6f2f67656e65726963',
+              MemoData: '72656e74',
+            },
+          },
+        ],
+      }
+
+      const result = Wallet.fromSeed(secret).signTransaction(lowercaseMemoTx)
+      assert.equal(
+        result,
+        '120000228000000023000022B8240000000C2E0000270F201B00D5A36761400000000098968068400000000000000C73210305E09ED602D40AB1AF65646A4007C2DAC17CB6CDACDE301E74FB2D728EA057CF744730450221009C00E8439E017CA622A5A1EE7643E26B4DE9C808DE2ABE45D33479D49A4CEC66022062175BE8733442FA2A4D9A35F85A57D58252AE7B19A66401FE238B36FA28E5A081146C1856D0E36019EA75C56D7E8CBA6E35F9B3F71583147FB49CD110A1C46838788CD12764E3B0F837E0DDF9EA7C1F687474703A2F2F6578616D706C652E636F6D2F6D656D6F2F67656E657269637D0472656E74E1F1',
+      )
+    })
+
+    it('EscrowExecution', async function () {
+      const secret = 'snoPBrXtMeMyMHUVTgbuqAfg1SUTb'
+      const result = Wallet.fromSeed(secret).signTransaction(
+        JSON.parse(REQUEST_FIXTURES.escrow.txJSON) as unknown as Transaction,
+      )
+      assert.deepEqual(result, RESPONSE_FIXTURES.escrow.signedTransaction)
+    })
+
+    it('signAs', async function () {
```

Comment:
If all of these tests are copied from `test/client/sign`, the testing of X-Addresses should be included